Intersect.Client.Core/Entities/Resource.cs

```diff
@@ -81,21 +104,54 @@ private void ReloadSpriteTexture()
             return;
         }
 
-        if (IsDead && Descriptor.Exhausted.GraphicFromTileset ||
-            !IsDead && Descriptor.Initial.GraphicFromTileset)
+        switch (_currentState?.TextureType)
         {
-            if (GameContentManager.Current.TilesetsLoaded)
-            {
-                Texture = Globals.ContentManager.GetTexture(Framework.Content.TextureType.Tileset, _sprite);
-            }
-            else
-            {
-                _waitingForTilesets = true;
-            }
+            case ResourceTextureSource.Tileset:
+                if (GameContentManager.Current.TilesetsLoaded)
+                {
+                    Texture = GameContentManager.Current.GetTexture(TextureType.Tileset, _sprite);
+                }
+                else
+                {
+                    _waitingForTilesets = true;
+                }
+                break;
+
+            case ResourceTextureSource.Resource:
+                Texture = GameContentManager.Current.GetTexture(TextureType.Resource, _sprite);
+                break;
+
+            case ResourceTextureSource.Animation:
+                if (_stateAnimation?.Descriptor?.Id == _currentState.AnimationId)
+                {
+                    return;
+                }
+
+                _stateAnimation?.Dispose();
+                _stateAnimation = null;
+
+                if (MapInstance is not { } mapInstance)
+                {
+                    return;
+                }
+
+                if (_animationDescriptor is not { } animationDescriptor)
+                {
+                    return;
+                }
+
+                var animation = mapInstance.AddTileAnimation(animationDescriptor, X, Y, Direction.Up);
+                if (animation is { IsDisposed: false })
+                {
+                    animation.InfiniteLoop = true;
+                }
```

Comment:
How is `IsDisposed` ever going to be true right after creating the animation instance?